lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,715 @@
+        ChunkedGelfDecoder::new(
+            self.chunked_gelf.timeout_millis,
+            self.chunked_gelf.pending_messages_limit,
+        )
+    }
+}
+
+/// Options for building a `ChunkedGelfDecoder`.
+#[configurable_component]
+#[derive(Clone, Debug, Derivative, PartialEq, Eq)]
+pub struct ChunkedGelfDecoderOptions {
+    /// The timeout, in milliseconds, for a message to be fully received. If the timeout is reached, the
+    /// decoder drops all the received chunks of the incomplete message and starts over.
+    #[serde(
+        default = "default_timeout_millis",
+        skip_serializing_if = "vector_core::serde::is_default"
+    )]
+    pub timeout_millis: u64,
+
+    /// The maximum number of pending incomplete messages. If this limit is reached, the decoder starts
+    /// dropping chunks of new messages. This limit ensures the memory usage of the decoder's state is bounded.
+    #[serde(
+        default = "default_pending_messages_limit",
+        skip_serializing_if = "vector_core::serde::is_default"
+    )]
+    pub pending_messages_limit: usize,
+}
+
+impl Default for ChunkedGelfDecoderOptions {
+    fn default() -> Self {
+        Self {
+            timeout_millis: default_timeout_millis(),
+            pending_messages_limit: default_pending_messages_limit(),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct MessageState {
+    total_chunks: u8,
+    chunks: [Bytes; GELF_MAX_TOTAL_CHUNKS as usize],
+    chunks_bitmap: u128,
+    timeout_task: JoinHandle<()>,
+}
+
+impl MessageState {
+    pub const fn new(total_chunks: u8, timeout_task: JoinHandle<()>) -> Self {
+        Self {
+            total_chunks,
+            chunks: Self::default_chunks(),
+            chunks_bitmap: 0,
+            timeout_task,
+        }
+    }
+
+    pub const fn default_chunks() -> [Bytes; GELF_MAX_TOTAL_CHUNKS as usize] {
+        [const { Bytes::new() }; GELF_MAX_TOTAL_CHUNKS as usize]
+    }
+
+    pub fn is_chunk_present(&self, sequence_number: u8) -> bool {
+        let chunk_bitmap_id = 1 << sequence_number;
+        self.chunks_bitmap & chunk_bitmap_id != 0
+    }
+
+    pub fn add_chunk(&mut self, sequence_number: u8, chunk: Bytes) {
+        let chunk_bitmap_id = 1 << sequence_number;
+        self.chunks[sequence_number as usize] = chunk;
+        self.chunks_bitmap |= chunk_bitmap_id;
+    }
+
+    pub fn is_complete(&self) -> bool {
+        self.chunks_bitmap.count_ones() == self.total_chunks as u32
+    }
+
+    pub fn retrieve_message(&mut self) -> Option<Bytes> {
+        if self.is_complete() {
+            self.timeout_task.abort();
+            let chunks = &self.chunks[0..self.total_chunks as usize];
+            let mut message = BytesMut::new();
+            for chunk in chunks {
+                message.extend_from_slice(chunk);
+            }
+            Some(message.freeze())
+        } else {
+            None
+        }
+    }
+}
+
+#[derive(Debug, Snafu)]
+pub enum ChunkedGelfDecoderError {
```

Comment:
I added these variants for the error handling, instead of logging with `warn!`. Couldn't avoid the logging in the case of the timeouts, since they are done asynchronously.
Maybe the naming could be improved, any feedback?